### PR TITLE
Changing enum values to upper case snake case for consistency

### LINKED ISSUE #1246: Handle enums for detailType in `Details` Component
### Issue Description

The `Details` component in the `TaskDetails `module currently converts `detailType` values like `Started On` and `Ends On` to user-friendly labels `Started` and `Ended`. With the proposed change to use upper case snake case values like `STARTED_ON` and `ENDS_ON` for `detailType`, the `Details` component will no longer automatically convert these to the intended labels
### Expected Behavior

- The `Details` component should be updated to handle upper case snake case `detailType` values.
- The component should continue to display `Started` and `Ended` for `detailType` values `STARTED_ON` and `ENDS_ON` respectively.
- The update should not impact other functionalities or labels within the `Details` component.

### Current Behavior

- Currently, the `Details` component only recognizes and converts `Started On` and `Ends On` into `Started` and `Ended`.
- Switching to the upper case snake case will cause the labels to be displayed as `STARTED_ON` and `ENDS_ON`, which is not user-friendly.
- The changes need to be done in the [Details Component](https://github.com/Real-Dev-Squad/website-status/blob/develop/src/components/taskDetails/TaskDates.tsx#L35-L49)
- In our local env, GMT +00:00 is giving an error but in the server build, it is expecting GMT +05:30. Due to this time difference, we may face unexpected errors.

### Proposed Solution

- Modify the logic in the `Details` component to recognize upper case snake case values `STARTED_ON` and `ENDS_ON`.
- Move the enum values to a variable to avoid spelling mistakes.
- Implement the necessary changes to ensure the user-friendly labels `Started` and `Ended` are displayed correctly when the component receives upper case snake case `detailType` values.

### Test Scenarios to Include

- Verify that the `Details` component correctly converts `startedOn` to `Started` and `endsOn` to `Ended`.
- Please make sure that the tooltip functionality for showing the full date is not affected by this change.
- Test for any regressions in the rendering of other `detailType` values.

### Reproducibility

 - [x] This issue is reproducible
 - [ ] This issue is not reproducible

### Severity/Priority
 - [ ] Critical
 - [x] High
 - [ ] Medium
 - [ ] Low

### Additional Information

- Careful consideration will be given to avoid introducing any breaking changes in the `Details `component.
### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by iAmAshuSahoo**
Date: 07 Sep 2024

Developer Name: Ashutosh Sahoo

---

## Issue Ticket Number

- Closes #1246 

## Description

Update Details Component to Handle Camel Case detailType Values 

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

image

</details>

## Test Coverage
<details>
<summary>Screenshot 1</summary>

image
</details>

### CODE REVIEW

**Review by iamitprakash — APPROVED** [review]

**Review by Achintya-Chatterjee — DISMISSED** [review]

**Comment by iAmAshuSahoo** [comment]
In my local GMT +00:00 is giving error but in the build it is expecting GMT +05:30

image
image

Proposed test
`
        const dateElements = screen.getByText(
            /Monday, Apr 19, 2021, 12:00 AM GMT/
        );
        expect(dateElements).not.toBeNull();
`

File: src/components/taskDetails/TaskDates.tsx [hunk 1]
```diff
@@ -32,20 +32,23 @@ export const TaskDates: React.FC<TaskDatesProps> = ({
     return (
         <>
             <div className={styles.inputContainer}>
-                <Details detailType="Started On" value={startedOn} />
+                <Details
+                    detailType="startedOn"
+                    value={'3/30/2024, 11:20:00 AM'}
```
Comment (ankushdharkar) [inline src/components/taskDetails/TaskDates.tsx:37]: Why hardcode this @iAmAshuSahoo ?

Comment (iAmAshuSahoo) [reply src/components/taskDetails/TaskDates.tsx:38]: I am now maintaining the values in a constant file.

**Review by Achintya-Chatterjee — APPROVED** [review]

File: src/components/taskDetails/TaskDates.tsx (hunk 1 above)
Comment (samarpan1738) [reply src/components/taskDetails/TaskDates.tsx:38]: Why is this date still hardcoded? @iAmAshuSahoo

Comment (iAmAshuSahoo) [reply src/components/taskDetails/TaskDates.tsx:38]: As a part of this task, we needed to move enums in the constant file. Other values are used as BAU.

Comment (iAmAshuSahoo) [reply src/components/taskDetails/TaskDates.tsx]: I have fixed that.

**Review by samarpan1738 — APPROVED** [review]

**Review by iamitprakash — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

**Pull request merged**